### ISSUE TITLE
{NPC} Salvaged Demolisher

### ISSUE THREAD

**Issue by durrrr** [TrinityCore/TrinityCore#3831]
Hello,
I have encountered a problem while questing in "Ulduar" zone. 

How to reproduce the bug: 
1. Get inside the "Ulduar" Zone.
2. Mount up on a "Salvaged Demolisher".
3. Try to mount a 2nd player to the "Passenger" sit.

What should happen, what happens instead:
You should mount the "Salvaged Demolisher" and then have the possiblity to mount another player so he will load "Blue Pyrite's" on to the "Salvaged Demolisher's" .(notice the quest data) instead, when a passenger loads a "Blue Pyrite" the whole "Passenger" sit is removed.

Some links:
http://www.wowhead.com/npc=33109/salvaged-demolisher#abilities --> NPC Abilities.
http://www.wowhead.com/npc=33109 --> The NPC.
http://www.wowhead.com/zone=4273 --> The Ulduar Zone.

I'm running Unix on hash: 37abe2528c8efa4a99feca0bd0a8e0b5e6fa3efc , TDB 43+
Addons: AH bot and WG.

**Comment by Aokromes** [comment]
I think this bug was fixed long time ago.

**Comment by sirikfoll** [comment]
Still valid on rev: 747ea7ca2c2b
Please, reopen.

Currently, passenger is going to the seat 2 of the demolisher, that's wrong.
The passenger should enter in the http://www.wowhead.com/npc=33167, which is already installed in the seat 1 from the demolisher.

As for seat 2, it need to have a accessory installed, is easily solved by:
```sql
DELETE FROM `vehicle_template_accessory` WHERE `entry`=33109 AND `accessory_entry`=33620;
INSERT INTO `vehicle_template_accessory` (`entry`, `accessory_entry`, `seat_id`, `minion`, `description`, `summontype`, `summontimer`) VALUES
(33109, 33620, 2, 1, 'Salvaged Demolisher', 6, 30000);
```

If seat 2 is occupied by the acessory, the passenger will go to seat 3(the arm of the catapult), still wrong.

**Comment by Thomson512** [comment]
Still bugged, passenger do not have any abilities, only "leave" button on Demolisher, on Engine, there is not any button. Need single issue??

**Comment by Keader** [comment]
@Thomson512 It has no spells as it is on the wrong seat.
He is the "catapult seat" (spell load into catapult) and not in "weapon seat"

**Comment by Bullfface** [comment]
Any word on this issue?

**Comment by hlainin** [comment]
Maybe someone knows how to fix Salvaged Siege Engine when player 1 sits in the driver's seat and player 2 tries to sit on the second place, he gets to the place of player 1 and player 2 freezes

**Comment by xinef1** [comment]
Imo few steps should be done:
1) Create a function that checks solely for VEHICLE_SEAT_FLAG_CAN_ENTER_OR_EXIT flag to count usable by client seats
2) UsableSeatNum should be based on numer calculated in 1)
3) If we install an accessory that is vehicle, we should transfer usable by client seatnum to base vehicle
4) If player enters vehicle at seat x, we should check if this seat is not occupied by accessory which is vehicle, if it is we should enter the accessory and not original vehicle
5) Reduce usable seat num from original vehicle

If done properly, this will fix ulduar vehicles

**Comment by pangolp** [comment]
Good afternoon, was it possible to resolve the issue of the passenger in the vehicle?

**Comment by illfated** [comment]
@pangolp : No. This issue is still open because none of the commits marked this issue as solved.
A working solution is still needed.
